lib/galaxy/jobs/handler.py

```diff
@@ -337,7 +337,7 @@ def __check_jobs_at_startup(self):
         if self.sa_session.dirty:
             self.sa_session.flush()
 
-    def __recover_job_wrapper(self, job):
+    def __recover_job_wrapper(self, job, resubmit=False):
```

Comment:
Do we initialize the `JobDestination` correctly, or should we use `from_job`: 

https://github.com/galaxyproject/galaxy/blob/88b455c953ce1b445de2cb63fadb9fc019ba1e2d/lib/galaxy/jobs/__init__.py#L118

What is `JobDestination["params"]` storing? 

Do we need to restore more than `resubmit` and `env` (`url`, `tags`, `runner`, `legacy`, `converted`, `shell`)